torchci/rockset/metrics/__sql/completed_pr_jobs_aggregate.sql

```diff
@@ -0,0 +1,117 @@
+-- This query is used to generate the CI Wait Time KPI for the pytorch/pytorch repo
+-- It's not the full kpi. Rather, this performs some early data processing and aggregation, which
+-- is then used by a python script to generate the final KPI, which gets uploaded back to rockset 
+-- to be generally queryable
+WITH
+    -- Get all PRs that were merged into master, and get all the SHAs for commits from that PR which CI jobs ran against
+    -- We need the shas because some jobs (like trunk) don't have a PR they explicitly ran against, but they _were_ run against
+    -- a commit from a PR
+    pr_shas AS (
+        SELECT  
+            r.pull_requests[1].number AS pr_number,
+            CONCAT(
+                'https://github.com/pytorch/pytorch/pull/',
+                r.pull_requests[1].number
+            ) AS url,
+            j.head_sha AS sha,
+        FROM
+            commons.workflow_job j
+            INNER JOIN commons.workflow_run r on j.run_id = r.id
+        WHERE
+            1 = 1
+            AND j._event_time > (
+                CURRENT_DATETIME() - DAYS(:from_days_ago)
+            )
+            AND r._event_time > (
+                CURRENT_DATETIME() - DAYS(:from_days_ago)
+            )
+            AND j._event_time < (CURRENT_DATETIME() - DAYS(:to_days_ago))
+            AND r._event_time < (CURRENT_DATETIME() - DAYS(:to_days_ago))
+            AND LENGTH(r.pull_requests) = 1
+            AND r.head_branch NOT IN ('master', 'main', 'nightly')
+            AND r.pull_requests[1].head.repo.name = 'pytorch'
+            AND r.name IN ('pull', 'trunk', 'Lint') -- Ensure we don't pull in random PRs we don't care about
+            AND (
+                r.pull_requests[1].base.ref = 'master'
+                OR r.pull_requests[1].base.ref like 'gh/%/base'
+            )
+        GROUP BY
+            pr_number,
+            url,
+            sha
+    ),
+    -- Now filter the list to just closed PRs
+    merged_pr_shas AS (
+        SELECT  
+            DISTINCT s.pr_number,
+            s.url,
+            s.sha
+        FROM
+            pr_shas s
+            INNER JOIN commons.pull_request pr on s.pr_number = pr.number
+        WHERE
+            pr.closed_at IS NOT NULL 
+            -- Ensure the PR was actaully merged
+            AND 'Merged' IN (
```

Comment:
Just FYI, for this condition, you could take a look at the new `commons.merges` collection if checking for the `Merged` label is not accurate

Reply:
Good to know, thanks!